clang/include/clang/Basic/Builtins.td

```diff
@@ -833,6 +833,12 @@ def BuiltinVaStart : Builtin {
   let Prototype = "void(__builtin_va_list_ref, ...)";
 }
 
+def BuiltinC32VaStart : LangBuiltin<"C23_LANG"> {
```

Comment:
Actually KIND OF sad that this is C23 only.  a `va_start` macro having to change its implementation based on language version is sad.  I would think making this builtin available as an extension in all language modes would be defensible.

Reply:
This matches GCC's behavior: https://godbolt.org/z/K8Tdqa9v4

Reply:
still KIND OF sad.